src/pages/login/login.ts

```diff
@@ -21,38 +21,31 @@ export class LoginPage {
   user      : User;
 
   constructor(public navCtrl: NavController,
-              public navParams: NavParams,
               public authService : AuthServiceProvider,
               public loadingCtrl : LoadingController,
               public alertCtrl   : AlertController) {
   }
 
   login() {
     this.loginData = new Credentials(this.usrName, this.usrPsw);
-    this.user = this.authService.authUser(this.loginData);
-    this.waitForResponse().then(() => {
-      setTimeout(() => {
-
-        if(this.user){
-          this.navCtrl.setRoot(HomePage, this.user);
-        }
-        else{
-          this.incorrectData();
-        }
-      }, 3000);
-    });
-  }
-
-  waitForResponse(){
     let loading = this.loadingCtrl.create({
       content: 'Logging in, please wait...',
-      spinner: 'circles',
-      duration: 3000
+      spinner: 'circles'
     });
-
-    return loading.present();
+    loading.present();
+    this.authService.authUser(this.loginData).subscribe(
+      (res: Response) => {
+        loading.dismiss();
+        this.user = new User(res['message']['id_user'], res['message']['username']);
```

Comment:
I suggest this way to handle de response:
`res.message.id_user` or `res.message.username` without the [ ]